hq_superset/tasks.py

```diff
@@ -11,5 +11,12 @@ def refresh_hq_datasource_task(domain, datasource_id, display_name, export_path,
         refresh_hq_datasource(domain, datasource_id, display_name, export_path, datasource_defn, user_id)
     except Exception:
         AsyncImportHelper(domain, datasource_id).mark_as_complete()
+        _remove_export_file(export_path)
         raise
-    os.remove(export_path)
+    else:
```

Comment:
nit: should we consider using `finally` instead of `else` ?

Reply:
I was not sure finally would work since we are raising exception within except block again. let me check

Reply:
That did work, let me update.

Reply:
https://github.com/dimagi/commcare-analytics/pull/45/commits/a5afef5c56e26d5491324e0811943941c290a0f5

Reply:
took the opportunity to simplify and merge those commits. Overall change remains the same.